src/pages/workspaces/workspace/analysis/runtime-common.js

```diff
@@ -92,8 +92,9 @@ export const RuntimeStatusMonitor = ({ runtime, onRuntimeStoppedRunning = _.noop
 }
 
 export const AuthenticatedCookieSetter = () => {
-  const { registrationStatus } = useStore(authStore)
-  return registrationStatus === userStatus.registeredWithTos && getLocalPref(cookiesAcceptedKey) !== false ? h(PeriodicCookieSetter) : null
+  const { termsOfService } = useStore(authStore)
+  return (termsOfService.userAcceptedTos || termsOfService.isGracePeriodEnabled) &&
```

Comment:
Adding a note for later... Once #3671 is merged, `userAcceptedTos` should take the grace period into consideration and this can be simplified to just checking `termsOfService.userAcceptedTos`.